python/lsst/resources/http.py

```diff
@@ -188,7 +188,7 @@ def _send_expect_header_on_put() -> bool:
 
 
 @functools.lru_cache
-def _is_webdav_endpoint(path: Union[ResourcePath, str]) -> bool:
+def isWebdavEndpoint(path: Union[ResourcePath, str]) -> bool:
```

Comment:
This needs some though. I don't like it being a public API at all and it's only public because one test in butler uses it and I'm not sure if that test really should be doing so.

Reply:
The class `HttpResourcePath` exposes the (property) method `is_webdav_endpoint`. Maybe the test in butler can use that method. If that is the case, I would vote for the function `isWebdavEndpoint` not to be exposed.

Reply:
No. The user doesn't know what type of scheme they have so there should be no public properties that are specific to a particular scheme (so really it should be `_is_webdav_endpoint`). Code should not have to change for an S3 scheme for example.

Reply:
Maybe we call it `_is_webdav_endpoint` and use it in the butler test as a clue that it's a private attribute that can only be used in special scenarios? Then we'd still be noting that it's not a general ResourcePath feature whilst allowing us not to have to export the helper function

Reply:
> Maybe we call it `_is_webdav_endpoint` and use it in the butler test
That seems a reasonable compromise. Butler tests would need modification then.

Please confirm I should proceed renaming `isWebdavEndpoint` by `_is_webdav_endpoint`.

Reply:
Rename is fine. It's a minor patch of daf_butler (which would have to be done on the same ticket).

Reply:
Modification done and pull request for `daf_butler` [submitted](https://github.com/lsst/daf_butler/pull/652) (ticket DM-33769)